main.py

```diff
@@ -53,7 +53,131 @@ def handle_arguments(cl_arguments):
     return parser.parse_args(cl_arguments)
 
 
-def _try_logging_git_info():
+def setup_target_task_training(args, target_tasks, model):
+    """
+    Save model states from pretrianing if applicable, and 
+    loads the correct model state for the target task training 
+    stage.
+    
+    Parameters
+    ----------------
+    args: Params object
+    target_tasks: list of target Task objects
+    mdoel: a MultTaskModel object
+
+    Returns
+    ----------------
+    task_names_to_avoid_loading: list of strings, if we don't allow for 
+    use of pretrained target specific module parameters, then this list will 
+    cosnist of all the task names so that we avoid loading the 
+    pretrained parameters. Else, it will be an empty list.
+    strict: bool, indicates whether to exit if the loaded model has some expected
+    parameters missing
+
+    """
+    if not args.do_target_task_training:
+        log.info("In strict mode because do_target_task_training is off. "
+                 "Will crash if any tasks are missing from the checkpoint.")
+        strict = True
+    else:
+        strict = False
+
+    if args.do_target_task_training and not args.allow_reuse_of_pretraining_parameters:
+        # If we're training models for evaluation, which is always done from scratch with a fresh
+        # optimizer, we shouldn't load parameters for those models.
+        # Usually, there won't be trained parameters to skip, but this can happen if a run is killed
+        # during the do_target_task_training phase.
+        task_names_to_avoid_loading = [task.name for task in target_tasks]
+    else:
+        task_names_to_avoid_loading = []
+
+    if not args.load_eval_checkpoint == "none":
+        # This is to load a particular eval checkpoint.
+        log.info("Loading existing model from %s...", args.load_eval_checkpoint)
+        load_model_state(model, args.load_eval_checkpoint,
+                         args.cuda, task_names_to_avoid_loading, strict=strict)
+    else:
+        # Look for eval checkpoints (available only if we're restoring from a run that already
+        # finished), then look for training checkpoints.
+
+        if args.transfer_paradigm == "finetune":
+            # Save model so we have a checkpoint to go back to after each task-specific finetune.
+            model_state = model.state_dict()
+            model_path = os.path.join(args.run_dir, "model_state_untrained_prefinetune.th")
+            torch.save(model_state, model_path)
```

Comment:
Not something we have to address this PR, but something seems off here (likely my fault). We should only need to create this checkpoint if we don't find a best checkpoint later. If we do find a best checkpoint to load, we're extraneously creating this checkpoint.

Reply:
Add an issue if not fixing now?

Reply:
done

Reply:
Can you make sure to merge master into this? This was fixed in another earlier PR.